utils/cligen/Makefile

```diff
@@ -0,0 +1,66 @@
+#
+# Copyright (C) 2020-2023 Olof Hagsand and Rubicon Communications, LLC(Netgate)
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=cligen
+PKG_VERSION:=6.4.0
+PKG_RELEASE:=1
+PKG_MAINTAINER:=Olof Hagsand <olof@hagsand.se>, Philip Prindeville <philipp@redfish-solutions.com>
+PKG_LICENSE:=Apache-2.0
+PKG_LICENSE_FILES:=LICENSE.md
+
+PKG_SOURCE_URL:=https://github.com/clicon/$(PKG_NAME)/archive/refs/tags
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL_FILE:=$(PKG_VERSION).tar.gz
```

Comment:
Why? This does not look good.

Reply:
Because the `PKG_NAME` doesn't occur in the tarball name on GitHub, since these are tags, not releases.  When we download it then, we need to make sure it gets put into a file which includes the `PKG_NAME` or `cligen` and `clixon` with clash as they'll both be called `6.4.0.tar.gz`.

Reply:
Hey man, c'mon. This is wrong. Don't give it even a try to convince me that there is no better way. Of course, there is. :) This is not the only package, which is using tags. Thus we don't need to reinvent a wheel once again. 

For example, if we want to use codeload:
https://codeload.github.com/clicon/cligen/tar.gz/6.4.0?

It will look in Makefile like this:
```
PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
PKG_SOURCE_URL:=https://codeload.github.com/clicon/cligen/tar.gz/V$(PKG_VERSION)?
PKG_HASH:=sha256sum
```

If you don't want to use codeload, then it's fine. Feel free to look at OpenWrt documentation or you can try to use some Makefile as an example in this repository.

Reply:
I hadn't used `codeload` before, surprisingly.  Done.